src/ucs/datastruct/mpool.c

```diff
@@ -265,6 +272,18 @@ void ucs_mpool_grow(ucs_mpool_t *mp, unsigned num_elems)
 void *ucs_mpool_get_grow(ucs_mpool_t *mp)
 {
     ucs_mpool_data_t *data = mp->data;
+    size_t max_chunk_size, chunk_size;
+    unsigned num_elems;
+
+    if (data->grow_factor != 1 && data->chunks != NULL) {
```

Comment:
why need this check?
can't we always multiply by grow_factor; if it's 1 it will have no effect

Reply:
as most cases grow_factor=1, so made this check to optimize/skip some lines of code.

Reply:
this is not significant because ucs_mpool_get_grow() is a slow operation anyway that happens rarely
better simplify the code

Reply:
removed checking grow_factor=1